app/services/plugin.service.ts

```diff
@@ -1,236 +1,61 @@
-
-    constructor(definitions: Definitions) {
-        this.nodes = definitions.nodes;
-
-        for (const [edgeType, values] of definitions.edges.entries()) {
-            const [st, dt] = [values.typeOf("source"), values.typeOf("destination")];
-            if (!((st instanceof ClassMetaType) && (dt instanceof ClassMetaType))) {
-                throw "Validator.constructor: inconsistancy error";
-            }
-            this.edges.set(edgeType, [st, dt]);
-        }
-    }
-
-    isValidEdge(t: string, src: string, dst: string): boolean {
-        const edgets = this.edges.get(t);
-        const srct = this.nodes.get(src);
-        const dstt = this.nodes.get(dst);
-        if (!edgets || !srct || !dstt) {
-            throw "validator state error";
-        }
-        const [srce, dste] = edgets;
-        return srct.subtype(srce) && dstt.subtype(dste);
-    }
-}
-
-class ConcretePlugin implements Core.Plugin {
-    kind = MagicConstants.DFA_PLUGIN_KIND;
-
-    get nodeTypes() {
-        return this.definitions.nodes.keys();
-    }
-    get edgeTypes() {
-        return this.definitions.nodes.keys();
-    }
-
-    validator: Validator;
-
-    constructor(private definitions: Definitions, public script: Script) {
-        this.validator = new Validator(definitions);
-    }
-
-    graphPluginData(kind: string) {
-        const type = this.definitions.graphs.get(kind);
-        if (!type) {
-            throw "type not found";
-        }
-        return new PluginPropertyData(kind, type);
-    }
-    nodePluginData(kind: string) {
-        const type = this.definitions.nodes.get(kind);
-        if (!type) {
-            throw "type not found";
-        }
-        return new PluginPropertyData(kind, type);
-    }
-    edgePluginData(kind: string) {
-        const type = this.definitions.edges.get(kind);
-        if (!type) {
-            throw "type not found";
-        }
-        return new PluginPropertyData(kind, type);
-    };
+export declare class Program {
+    constructor(any: SerialJSO);
+    run(a: any): any;
 }
+type StubContext = { global: { "plugin-stub": { "Program": typeof Program } } };
 
 @Injectable()
 export class PluginService {
-    private plugins = new Map<string, Promise<ConcretePlugin>>();
-    private interpretCode: Script;
-    private runInputCode: Script;
+    private plugins = new Map<string, Plugin>();
+    private programs = new Map<Plugin, Promise<StubContext>>();
+    private getResults: Script;
+    private addGraph: Script;
     // TODO: load from somewhere
-    private pluginKinds = new Map([[MagicConstants.DFA_PLUGIN_KIND, { definitions: "./dfa-definition.sinapdef", interpreter: "./build/plugins/dfa-interpreter.js" }]])
+    private pluginKinds = new Map([[MagicConstants.DFA_PLUGIN_KIND, "./plugins/dfa-interpreter.ts"]])
 
-    constructor( @Inject(LocalFileService) private fileService: FileService,
+    constructor( @Inject(LocalFileService) private fileService: LocalFileService,
```

Comment:
According to what I read about Angular depedency injection, even though the @Inject is a specific type, we can change it to be something else that fulfills the FileService interface.